qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py

```diff
@@ -19,7 +19,14 @@
 
 import numpy as np
 
-from retworkx import NodeIndices, PyGraph, WeightedEdgeList, adjacency_matrix
+try:
+    import networkx as nx
+
+    HAS_NX = True
+except ImportError:
+    HAS_NX = False
```

Comment:
I guess you can refactor this in-line with #535

Reply:
To be really lazy, ideally the import  should be moved inside the method it is used and the method marked with:
```
@_optionals.HAS_NETWORKX.require_in_call
```
If that is not possible then the non lazy option but still complying with the new Terra API is to change the top with:
```
if _optionals.HAS_NETWORKX:
    # pylint: disable=unused-import
    import networkx as nx
```

And in the method where you would check the HAS_NX, just add:
```
@_optionals.HAS_NETWORKX.require_in_call
```
or if it is a `init` call, use it in the class definition:
```
@_optionals.HAS_NETWORKX.require_in_instance
```

Reply:
Ok, then I'll wait for #535 to be merged 👍🏻 Thanks Manoel!

Reply:
In this case the `HAS_NETWORKX` is from Terra so you don't need the Nature changes but they have been merged.